tectonic/dpx-dvipdfmx.c

```diff
@@ -400,6 +403,12 @@ dvipdfmx_main (
     tt_aux_set_verbose(verbose);
   }
 
+  if (compress) {
+    pdf_set_compression(9);
+  } else {
+    pdf_set_compression(0);
+  }
+
```

Comment:
Maybe just `pdf_set_compression(compress ? 9 : 0)` ? One of the few cases that is concise enough that I don't find the ternary operator to add any confusion.